src/NuGet.Protocol.Catalog/Models/PackageVulnerability.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Newtonsoft.Json;
+
+namespace NuGet.Protocol.Catalog
+{
+    public class PackageVulnerability
+    {
+        [JsonProperty("@id")]
+        public string Id { get; set; }
+
+        [JsonProperty("@type")]
+        public string Type { get; set; }
+
+        [JsonProperty("advisoryUrl")]
+        public string AdvisoryUrl { get; set; }
+
+        [JsonProperty("gitHubDatabaseKey")]
+        public string GitHubDatabaseKey { get; set; }
```

Comment:
Is this property supposed to be included? The OneNote spec only mentions `severity`, `advisoryUrl`, and `@type` (see [this](https://microsoft.sharepoint.com/teams/NuGet/_layouts/OneNote.aspx?id=%2Fteams%2FNuGet%2FTeam%2FNugetServer%2FNugetServerTeamNote&wd=target%28Specs%20%28in%20progress%5C%29.one%7C910181C5-8695-4A1C-9792-A56DBB86F1C3%2FV3%20Protocol%20-%20Exposing%20Vulnerability%20information%7CE859AEBF-5731-4AAA-8066-DF8617F464C4%2F%29
onenote:https://microsoft.sharepoint.com/teams/NuGet/Team/NugetServer/NugetServerTeamNote/Specs%20(in%20progress).one#V3%20Protocol%20-%20Exposing%20Vulnerability%20information&section-id={910181C5-8695-4A1C-9792-A56DBB86F1C3}&page-id={E859AEBF-5731-4AAA-8066-DF8617F464C4}&object-id={DA482374-BD34-4266-B41F-5518BC39241F}&5D)).

Reply:
It seems odd not to add it, as it would be useful metadata to have if we want to provide a link in the PM UI, for example. As well as your thoughts, @loic-sharma, I'd like @xavierdecoster and @joelverhagen on this one. Thoughts?

Reply:
I'll try to take a look at the rest of this PR tomorrow. Sorry about the delay. Regarding this particular issue, I recommend putting the bare minimum that we need for the client experience. If there is no scenario for this field in registration, we can exclude it. 

Catalog is immutable so it's trickier to change later (reflow all the things). For registration, we simply (...) do a rebuild and we can add the field later if we need it.

I agree with @loic-sharma. In fact, I would question even the `@type`. This V3 protocol spec was written before the Catalog2Registration rebuild so it would have been hard to not do the `@type`. Perhaps that was the consideration. Now it's very easy to do the minimal properties so I think we should consider dropping all but `severity` and `advisoryUrl`. The `@type` and `@id` made sense for generic graph processing a la JSON-LD and RDF. The reality is that the readers and writers of this JSON are very likely strongly typed code with models, etc. Client uses Newtonsoft.Json in a similar way to how this is written.

This brings up the question... has the client experience been designed? Do we know what we want to show on the client experience? How do we even know `advisoryUrl` and `severity` are needed? Or perhaps there is more needed?

Let's be sure the cart doesn't go before the horse (or is it the other way around...)

Reply:
Sold. :) I'll include this change.

Reply:
> has the client experience been designed? Do we know what we want to show on the client experience? How do we even know advisoryUrl and severity are needed? Or perhaps there is more needed?

This is a great point. We avoid breaking changes in the V3 protocol, so we shouldn't update the protocol until we understand the client experience. Should we block this PR until that's ironed out?

Reply:
The client experience indicates a need for the advisory URL and the severity of the vulnerability:
* In the [dotnet CLI spec](https://github.com/NuGet/Home/blob/dev/designs/PackageVulnerability/DotnetListPackageVulnerable.md) we see a simple flagging of vulnerabilities with a (V), along with the advisory URL
* In the [PM UI spec](https://www.figma.com/file/8gCQx9iAQfPXTtxoks6Qd7/%F0%9F%9B%A1%EF%B8%8F-NuGet-Vulnerable-Packages) we can see that a vulnerable package will be flagged as such, with two hyperlinks: "more details" (advisory URL) and "fix" (seek out the first patched version, which may need to be added to metadata). I'll check in with @anangaur again on this detail.